master/internal/stream/authz_iface.go

```diff
@@ -0,0 +1,21 @@
+package stream
+
+import (
+	"context"
+
+	"github.com/determined-ai/determined/master/internal/authz"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// StreamAuthZ is the interface for streaming authorization.
+type StreamAuthZ interface {
+	// GetTrialStreamableScopes returns a AccessScopeSet where the user has permission to view trials.
+	GetTrialStreamableScopes(ctx context.Context, curUser model.User) (model.AccessScopeSet, error)
+
+	// WaitForPermissionChange returns an error whenever
+	// a change that modifies permission scopes are detected.
+	WaitForPermissionChange() error
```

Comment:
Also, do we need this function at all?  You could also just configure the NOTIFY listener to call PublisherSet.Reset() or whatever, right?

Reply:
(though I'm not sure if that's what's best)

Reply:
Swapped this out for getting permission listener